pandas/tests/frame/methods/test_compare.py

```diff
@@ -238,17 +238,50 @@ def test_invalid_input_result_names(result_names):
         df1.compare(df2, result_names=result_names)
 
 
-def test_compare_ea_and_np_dtype():
-    # GH#44014
-    df1 = pd.DataFrame({"a": [4.0, 4], "b": [1.0, 2]})
-    df2 = pd.DataFrame({"a": pd.Series([1, pd.NA], dtype="Int64"), "b": [1.0, 2]})
-    result = df1.compare(df2, keep_shape=True)
+@pytest.mark.parametrize(
+    "ea_val,np_dtype_val",
```

Comment:
these names are confusing, can you clarify

Reply:
`ea_val` and `np_dtype_val` intended to communicate value for extension array and value for np dtype array respectively.

is this ok?
```suggestion
    "df1_val,df2_val",
```

Reply:
yep this is clearer